ci/actions/dev-build-tag-gen.sh

```diff
@@ -128,21 +134,38 @@ last_tag=""
 version_tags=""
 previous_release_major=0
 previous_release_minor=0
+# If option -r is set, then the script will handle tag generation as for the release branch, it'd do 'develop' branch
+# otherwise.
 if [[ $previous_release_gen == false ]]; then
+#   List of all beta-build (DB) tags in reverse version order.
     version_tags=$(git tag | sort -V -r | grep -E "^(V(${current_version_major}).(${current_version_minor})(DB[0-9]+))$" || true)
     last_tag=$(get_first_item "$version_tags")
 else
     previous_release_major=$(( current_version_major - 1 ))
-    version_tags=$(git tag | sort -V -r | grep -E "^(V(${previous_release_major}).([0-9]+)(DB[0-9]+)?)$" || true)
+#   List of official release tags in reverse order.
+    version_tags=$(git tag | sort -V -r | grep -E "^(V(${previous_release_major}).([0-9]+))$" || true)
```

Comment:
I think the dot should be escaped to ensure it matches a for and not any character